__tests__/helpers/url.spec.js

```diff
@@ -47,6 +47,18 @@ describe('buildFetchUrl', () => {
     expect(url).toBe('http://api.domain.com/users/123/posts');
   });
 
+  it('should handle empty optional ids by removing extra forward slashes or question marks', () => {
+    const baseUrl = 'http://api.domain.com/users/:userId?/posts/:postId';
```

Comment:
Can you add a test with multiple optional fields?

Reply:
Thanks for pointing this out. I had to add the global search flag to the regex in `buildFetchUrl` so it looks like `/\?\//g`. In order to get it to work after adding a test with multiple empty optional ids.